api/src/db-connection.ts

```diff
@@ -112,6 +112,31 @@ export class Connection {
     return isNaN(version) ? null : version;
   }
 
+  async getLatestProcessedArxivVersion(paperSelector: PaperSelector): Promise<number> {
+    if (isS2Selector(paperSelector)) {
+      // TODO (mjlangan): Error, or maybe null, think about this
+      return -1;
+    }
+    // Provided arXiv IDs might have a version suffix, but ignore that for this check.
+    const versionDelimiterIndex = paperSelector.arxiv_id.indexOf('v');
+    const arxivId = versionDelimiterIndex > -1 ? paperSelector.arxiv_id.substring(0, versionDelimiterIndex) : paperSelector.arxiv_id;
+
+    const response = await this._knex.raw<{ rows: { arxiv_version: number }[] }>(`
+      SELECT CAST((REGEXP_MATCHES(arxiv_id,'^\\d{4}\\.\\d{5}v(\\d+)$'))[1] AS integer) AS arxiv_version
+        FROM paper
+        WHERE arxiv_id ilike ?
+        ORDER BY arxiv_version DESC
```

Comment:
will this ordering handle double-digit version numbers well? For instance, will it order them like:

```v11, v10, v9, ... v2, v1``` (numerical)

or 

```v9, v8, ... v11, v10, v1``` (alphabetical) ?

Reply:
It should, since we're treating the extracted versions as integers.

Reply:
oh yes! you're absolutely right.